parquet/src/test/java/org/apache/iceberg/parquet/TestParquetDataWriter.java

```diff
@@ -112,5 +159,10 @@ public void testDataWriter() throws IOException {
     }
 
     Assert.assertEquals("Written records should match", records, writtenRecords);
+
+    Assert.assertTrue("Should have a valid lower bound", dataFile.lowerBounds().containsKey(1));
+    Assert.assertTrue("Should have a valid upper bound", dataFile.upperBounds().containsKey(1));
+    Assert.assertTrue("Should have a valid lower bound", dataFile.lowerBounds().containsKey(2));
```

Comment:
It seems UnicodeUtil and BinaryUtil truncateMin functions do not fail (they just truncate to the length and return).  

I hope it is ok to have a lowerBound but no upperBound for this column?

Reply:
I think we always do a null check before comparing, but lets check the metrics evaluators

Reply:
https://github.com/apache/iceberg/blob/bf9a22738b1160749b113f9b5f090a5e8d4c3827/api/src/main/java/org/apache/iceberg/expressions/InclusiveMetricsEvaluator.java#L231-L237

https://github.com/apache/iceberg/blob/bf9a22738b1160749b113f9b5f090a5e8d4c3827/api/src/main/java/org/apache/iceberg/expressions/InclusiveMetricsEvaluator.java#L256-L263 <- Yep should be safe

Reply:
Basically this means if the upper bound is missing (due to incalculable truncate) we just always fall back to "RowsMayMatch"